endpoint login

wojbog opened the pull request:
create new login endpoint

wojbog committed:
endpoint login

create new endpoint login

akruszewski commented on handlers/LoginUser.go:
```diff
@@ -0,0 +1,62 @@
+				"error":   "Cannot read request"})
+		}
+		//check in datebase and create token
+		if user, err := s.LoginUser(c.Context(), p); err != nil {
```
Tutaj tez moze byc internal server error, takze sprawdzalbym czy ten blad jest typu np. InvalidEmailOrPasswordError albo inny.

akruszewski commented on service/service.go:
```diff
@@ -45,6 +46,27 @@ func (s *Service) AddNewUser(ctx context.Context, user models.NewUser)(models.Us
+func (s *Service) LoginUser(ctx context.Context, user models.UserLogin)(models.UserLogin,error) {
+
+		//chek in datebase
+		if userDB,err:=s.userCol.GetUserLogin(ctx,user);err!=nil {
```
Formatowanie jest off, skonfiguruj sobie edytor, zeby to sam ogarnial.

akruszewski commented on repository/user/user.go:
```diff
@@ -41,6 +42,16 @@ func (colUser *Collection) InsertUser(ctx context.Context, user models.NewUser)
 	}
 }
+//GetUserLogin check in datebase
+func (colUser *Collection) GetUserLogin(ctx context.Context, user models.UserLogin) (models.UserLogin, error) { 
```
nazwalbym to GetUserByEmail i zwracal calego usera z bazy, moze sie przydac w innych miejscach.

akruszewski commented on models/models.go:
```diff
@@ -21,4 +25,10 @@ type UserResponse struct {
\ No newline at end of file
+}
+
+type UserLogin struct {
```
raczej dalbym to jako User, i umiescil wszystkie dane danego uzytkownika w strukturze.

akruszewski commented on handlers/LoginUser_test.go:
```diff
@@ -0,0 +1,13 @@
+)
+
+//TestCreateTokenReturnTrueIfFuncReturnToken test func CreateTokenR
+func TestCreateTokenReturnTrueIfFuncReturnToken(t *testing.T) {
```
A co z sad pathami?

shhhQuiettt committed:
Create ValidationError

createUser with validation error returns invalid fields in specific error field now

wojbog committed:
fix bug endpoint login

add second test func CreateToken

wojbog committed:
Merge remote-tracking branch 'origin/login' into login

akruszewski commented on handlers/LoginUser.go:
```diff
@@ -0,0 +1,71 @@
+		}
+		//check in datebase and create token
+		if user, err := s.LoginUser(c.Context(), p); err != nil {
+			if err == errors.New("incorrect password") || err == errors.New("incorrect data") {
```
To nie pyknie, stworz zmienna `var InvalidPasswordError = errors.New("incorrect password")` i tak samo dla InvalidDataError w pliku w ktorym rzucasz te errory, i sprawdz je tutaj.

wojbog committed:
add tests

add handlers test, repository test, service test
add errors
